WhatsApp dropdown update (#64)
botpress/documentation: pages/cloud/channels/whatsapp.mdx

**Special behaviors for choices, cards, and carousels**

Choices (buttons), dropdowns, cards, and carousels are all mapped to [Interactive List Objects.](https://developers.facebook.com/docs/whatsapp/cloud-api/reference/messages#interactive-object)

This comes with a few restrictions and behaviors specific to WhatsApp:
- When there are 3 or fewer choices as buttons or in a dropdown, they will be rendered as buttons.
- When there are more than 3 choices as buttons or in a dropdown, they will be rendered as a dropdown.
- If there are more than 10 choices in a dropdown, the multiple dropdown messages will be sent with up to 10 choices in each messsage.
  - For example, if your bot has a dropdown with 10 choices, the user will receive two messages: the first message will have a dropdown with 10 choices, and the second messge will have a dropdown with two choices.
- Each dropdown requires a label but the Studio currently doesn’t support that, so the label “Choose…” is used for all dropdowns.
- Button labels are limited to 20 characters, and dropdown labels are limited to 24 characters.

>   - Labels exceeding these limits are automatically truncated. For example, the label "This button label is very long" would become "This button label is" on a button and "This button label is ver" on a dropdown.

Reviewer: An ellipsis character is added at the end now when the label is truncated.

```suggestion
  - Labels exceeding these limits are automatically truncated. For example, the label "This button label is very long" would become "This button label i…" on a button and "This button label is ve…" on a dropdown.
```

> - There is a limit of 3 url action buttons on each cards

Reviewer: ```suggestion
- There is a limit of 3 URL action buttons on each cards
```

> - Each card in a carousel will be sent individually

Reviewer: This reminded me that there's another behavior around cards that a user recently asked about on Discord, so added a sentence at the bottom to clarify the behavior around this other Whatsapp limitation.

```suggestion
- Each card in a carousel will be sent individually
- Whatsapp requires cards to be "reply" buttons, and doesn't support URLs. If the card only has URL actions, a card won't be able to be displayed and it will be downrendered using normal text/image messages.
```

Reviewer: Yes, exactly

> 

Reviewer: We may also want to add here another aspect I recently investigated:

```suggestion
- When files are sent over Whatsapp, the Studio doesn't store the original filename (for security reasons) but only the file extension, so the file will be sent with a generic filename ("file") followed by the extension of the file. For example, if you send a PNG file, it will be sent as "file.png".
```

- Video is mapped to [Media Video Object](https://developers.facebook.com/docs/whatsapp/cloud-api/reference/messages#media-object)
- File is mapped to [Media File Object](https://developers.facebook.com/docs/whatsapp/cloud-api/reference/messages#media-object)
- Location is mapped to [Location Object](https://developers.facebook.com/docs/whatsapp/cloud-api/reference/messages#location-object)

**Special behaviors for choices, cards, and carousels**

> Choices (buttons), dropdowns, cards, and carousels are all mapped to [Interactive List Objects.](https://developers.facebook.com/docs/whatsapp/cloud-api/reference/messages#interactive-object)

Reviewer: Might be better to move this one a bit above back where it was with the other links, for better visibility.

This comes with a few restrictions and behaviors specific to WhatsApp:
- When there are 3 or fewer choices as buttons or in a dropdown, they will be rendered as buttons.
- When there are more than 3 choices as buttons or in a dropdown, they will be rendered as a dropdown.
- If there are more than 10 choices in a dropdown, the multiple dropdown messages will be sent with up to 10 choices in each messsage.
  - For example, if your bot has a dropdown with 10 choices, the user will receive two messages: the first message will have a dropdown with 10 choices, and the second messge will have a dropdown with two choices.
- Each dropdown requires a label but the Studio currently doesn’t support that, so the label “Choose…” is used for all dropdowns.
- Button labels are limited to 20 characters, and dropdown labels are limited to 24 characters.
  - Labels exceeding these limits are automatically truncated. For example, the label "This button label is very long" would become "This button label i…" on a button and "This button label is ve…" on a dropdown.
- There is a limit of 3 URL action buttons on each cards

> - Each card in a carousel will be sent individually

Reviewer: ```suggestion
- Each card in a carousel will be sent individually.
```

> - Whatsapp requires cards to be "reply" buttons, and doesn't support URLs. If the card only has URL actions, a card won't be able to be displayed and it will be downrendered using normal text/image messages.

Reviewer: ```suggestion
- Whatsapp requires cards to only use "reply" buttons, and doesn't support URLs. If the card only has URL actions, a card won't be able to be displayed and it will be downrendered using normal text/image messages.
```
